Core/include/Acts/Utilities/Intersection.hpp

```diff
@@ -198,22 +198,17 @@ namespace detail {
 template <typename intersection_t, typename logger_t = std::false_type>
 bool checkIntersection(const intersection_t& intersection, double pLimit,
                        double oLimit, double tolerance,
-                       [[maybe_unused]] logger_t logger = logger_t{}) {
-  constexpr bool doLogging = not std::is_same_v<logger_t, std::false_type>;
-
+                       const Logger& logger = getDummyLogger()) {
   const double cLimit = intersection.pathLength;
 
-  if constexpr (doLogging) {
-    ACTS_VERBOSE(" -> pLimit, oLimit, cLimit: " << pLimit << ", " << oLimit
-                                                << ", " << cLimit);
-  }
+  ACTS_VERBOSE(" -> pLimit, oLimit, cLimit: " << pLimit << ", " << oLimit
+                                              << ", " << cLimit);
 
   const bool coCriterion = cLimit > oLimit;
   const bool cpCriterion = std::abs(cLimit) < std::abs(pLimit) + tolerance;
 
   const bool accept = coCriterion and cpCriterion;
 
-  if constexpr (doLogging) {
     if (accept) {
       ACTS_VERBOSE("Intersection is WITHIN limit");
     } else {
```

Comment:
I guess this and the following should be moved to the left?

Reply:
Huh, clang-tidy doesn't complain?

Reply:
Nevermind, it does.